enginetest/join_planning_tests.go

```diff
@@ -448,6 +448,21 @@ order by 1;`,
 			"insert into uv values (0,1), (1,1), (2,2), (3,2);",
 		},
 		tests: []JoinPlanTest{
+			{
```

Comment:
How about table aliases? It looks like the code would need more logic to understand that a table name and it's aliased name are the same; doesn't seem like we necessarily need that for the first pass, but I'm wondering if some tests with aliases are still interesting.

Reply:
Totally not in scope for this PR either, but I was also thinking about `having` clauses and wondering if it's eventually worth doing a similar treatment, or being able to convert between where and having.

Reply:
Do you think it's worth adding any tests with plan output for any of these? I know the main point is to turn this from an inefficient subquery execution to a join plan, and we do assert that we have the expected join types in the plan. I don't have a strong opinion on this, but just wondering if we are testing deeply enough. 

Just brainstorming, not pushing for this... Do we have any tests that shuffle the plan around and then generate the corresponding, rewritten SQL statement for the new plan? Might be a little fragile, but could be a direct way to test the rewriting of the statement without the full plan output.

Reply:
1) I added a blocker for hoisting any filters with alias references. I still cannot trigger that condition right now, though. A HAVING->IN_SUBQUERY->WHERE->IN_SUBQUERY is the only way I can figure out how to reach that condition, and I don't recurse HAVING expressions right now. That is something we can optimize in the future.

2) re: table aliases, I am not sure I understand. We currently gather all named relational nodes in a scope, and those names are considered in-scope. If there is an out-of-scope table of the same name, get fields in the current scope will resolve to the in-scope relation.

3) Doing more HAVING transforms and generally converting HAVING->WHERE is interesting, but not a heavily reported case by customers yet, and that we can optimize separately.

4) A better IR for comparing transformations is appealing. SQL strings are a difficult format for output comparisons because not all of our operators have SQL token equivalents. CRDB has a couple flavors of lisp with some simple syntax highlighting for different types of transformation tests. I'm not against experimenting, but might stick with the presence of join operators for now.